libs/langchain/langchain/llms/fireworks.py

```diff
@@ -66,43 +70,89 @@ def _llm_type(self) -> str:
         """Return type of llm."""
         return "fireworks"
 
-    def _call(
+    def _generate(
         self,
-        prompt: str,
+        prompts: List[str],
         stop: Optional[List[str]] = None,
         run_manager: Optional[CallbackManagerForLLMRun] = None,
         **kwargs: Any,
-    ) -> str:
-        """Run the LLM on the given prompt and input."""
-        params: dict = {
+    ) -> LLMResult:
+        """Call out to Fireworks endpoint with k unique prompts.
+        Args:
+            prompts: The prompts to pass into the model.
+            stop: Optional list of stop words to use when generating.
+        Returns:
+            The full LLM output.
+        """
+        params = {
             "model": self.model,
-            "prompt": prompt,
             **self.model_kwargs,
         }
-        response = completion_with_retry(
-            self, run_manager=run_manager, stop=stop, **params
-        )
+        sub_prompts = self.get_batch_prompts(params, prompts, stop)
+        choices = []
+        for _prompts in sub_prompts:
+            response = completion_with_retry_batching(
+                self, self.use_retry, prompt=_prompts, run_manager=run_manager, **params
+            )
+            choices.extend(response)
 
-        return response.choices[0].text
+        return self.create_llm_result(choices, prompts)
 
-    async def _acall(
+    async def _agenerate(
         self,
-        prompt: str,
+        prompts: List[str],
         stop: Optional[List[str]] = None,
         run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
         **kwargs: Any,
-    ) -> str:
-        """Run the LLM on the given prompt and input."""
+    ) -> LLMResult:
+        """Call out to Fireworks endpoint async with k unique prompts."""
         params = {
             "model": self.model,
-            "prompt": prompt,
             **self.model_kwargs,
         }
-        response = await acompletion_with_retry(
-            self, run_manager=run_manager, stop=stop, **params
-        )
+        sub_prompts = self.get_batch_prompts(params, prompts, stop)
+        choices = []
+        for _prompts in sub_prompts:
+            response = await acompletion_with_retry_batching(
+                self, self.use_retry, prompt=_prompts, run_manager=run_manager, **params
```

Comment:
`stop` is in params

Reply:
I am not sure of this, can you please double check? **params will only contain any arguments other than the named parameters... in this case I think it will exclude stop.

```python
def hello(x, stop, **params):
    print(params)

hello(x="x_value", stop="stop_value", foo="bar")
```

If you run the above, it prints out `{'foo': 'bar'}`

Reply:
See earlier note, I don't believe this is correct.